core/trino-spi/src/main/java/io/trino/spi/predicate/SortedRangeSet.java

```diff
@@ -815,10 +815,16 @@ public int hashCode()
     {
         int hash = lazyHash;
         if (hash == 0) {
-            hash = Objects.hash(type, Arrays.hashCode(inclusive));
+            // Adding to the hash: type, inclusive and low/high unbounded indication (if available)
+            if (sortedRanges.getPositionCount() > 0) {
+                hash = Objects.hash(type, Arrays.hashCode(inclusive), sortedRanges.isNull(0), sortedRanges.isNull(sortedRanges.getPositionCount() - 1));
```

Comment:
`sortedRanges.isNull(0), sortedRanges.isNull(sortedRanges.getPositionCount() - 1)` seems redundant with respect to the loop below, which includes all positions' nullness in the hash

Reply:
without that part the second Assert in the test fails. 
If any, I would remove the other addition

Reply:
`hash = hash * 31 + position + 1;` will do the trick

Reply:
Does the trick for that one, but if I'll add this one:
`assertThat(ValueSet.ofRanges(Range.lessThan(BIGINT, 1L)).hashCode()).isNotEqualTo(ValueSet.ofRanges(Range.range(BIGINT, -934573233491768040L, false, 1L, false)).hashCode());`
it will fail

Reply:
I just think the safest thing is to relate explicitly to the "bound-ness" of the ranges

Reply:
So the question is: what problem you are trying to solve?

Reply:
Avoid collisions on very simple predicates. 
I do agree the last example does not fit to "simple predicates" definition.
I don't really mind staying with `hash = hash * 31 + position + 1;`

But why do we think that adding the explicit notion for the bound-ness is the wrong thing to do?

Reply:
I'm not stating that it's wrong :)

Reply:
@findepi your call :)

Reply:
> without that part the second Assert in the test fails. If any, I would remove the other addition

i don't get it.
maybe per-position should be smarter